src/StreamRequest.jl

```diff
@@ -19,27 +21,18 @@ immediately so that the transmission can be aborted if the `Response` status
 indicates that the server does not wish to receive the message body.
 [RFC7230 6.5](https://tools.ietf.org/html/rfc7230#section-6.5).
 """
-abstract type StreamLayer{Next <: Layer} <: Layer{Next} end
-export StreamLayer
-
-function request(::Type{StreamLayer{Next}}, io::IO, req::Request, body;
-                 reached_redirect_limit=false,
-                 response_stream=nothing,
-                 iofunction=nothing,
-                 verbose::Int=0,
-                 kw...)::Response where Next
-
+function streamlayer(stream::Stream; iofunction=nothing, verbose=0, redirect_limit::Int=3, kw...)::Response
```

Comment:
This is our only client-side "handler" function that is _not_ a middleware; i.e. it actually takes the `stream`, sends the request and reads the response. So all other client layers can be seen as "wrapping" this layer that does the actual send/receive operation.